### PR TITLE
Add  support for ed25519 signing

### PR DESCRIPTION
**Pull request by rahulredd**
This PR adds support for ed25519 signing and verification. 

Address this issue -  #68

### CODE REVIEW

File: jwk.go [hunk 1]
```diff
@@ -379,6 +444,14 @@ func (key rawJSONWebKey) rsaPrivateKey() (*rsa.PrivateKey, error) {
 	return rv, err
 }
 
+func fromEdPrivateKey(ed ed25519.PrivateKey) (*rawJSONWebKey, error) {
+
```
Comment (csstaub) [inline jwk.go]: Remove extra whitespace here

File: shared.go [hunk 2]
```diff
@@ -68,6 +68,7 @@ var (
 
 // Key management algorithms
 const (
+	ED25519Key         = KeyAlgorithm("ED25519")
```
Comment (csstaub) [inline shared.go]: Why not just call this constant `ED25519`?

File: asymmetric.go [hunk 3]
```diff
@@ -438,6 +466,33 @@ func (ctx ecDecrypterSigner) decryptKey(headers rawHeader, recipient *recipientI
 
 	return josecipher.KeyUnwrap(block, recipient.encryptedKey)
 }
+func (ctx edDecrypterSigner) signPayload(payload []byte, alg SignatureAlgorithm) (Signature, error) {
+
```
Comment (csstaub) [inline asymmetric.go]: Remove extra whitespace

File: asymmetric_test.go [hunk 4]
```diff
@@ -142,6 +142,45 @@ func TestVectorsRSA(t *testing.T) {
 	}
 }
 
+func TestEd25519(t *testing.T) {
+
```
Comment (csstaub) [inline asymmetric_test.go]: Remove extra whitespace

File: jwk.go [hunk 5]
```diff
@@ -126,6 +132,16 @@ func (k *JSONWebKey) UnmarshalJSON(data []byte) (err error) {
 		}
 	case "oct":
 		key, err = raw.symmetricKey()
+	case "OKP":
+		if raw.Crv == "Ed25519" {
```
Comment (csstaub) [inline jwk.go]: Could add a check here that X is present, as it must always be present for both private and public keys:
```
if raw.Crv == "Ed25519 && raw.X != "" {
        // parse 
} else {
        // return error
}
```

**Comment by csstaub** [comment]
Hi! Thanks for your contribution! This looks great, I left some minor comments. One thing I think we should add is more test vectors (ideally from the appendix of the spec) to make sure the code generates the correct output given a fixed set of parameters.

**Comment by rahulredd** [comment]
Fixed all the indentation issues and added test vectors.

**Comment by csstaub** [comment]
This looks good. Thank you for your contribution @rahulredd!

**Comment by csstaub** [comment]
If you want me to cut a new release with this, let me know.

**Pull request merged**